src/dimension/mod.rs

```diff
@@ -394,6 +382,19 @@ fn to_abs_slice(axis_len: usize, slice: Slice) -> (usize, usize, isize) {
     (start, end, step)
 }
 
+/// The offset value of the data pointer relative to the actual pointer of the given
+/// dim and strides. The result is always <= 0.
+pub fn head_ptr_offset(dim: &[Ix], strides: &[Ix]) -> isize{
+    let offset=izip!(dim, strides)
+        .fold(0,|_offset, (d, s)|{
+            if (*s as isize) < 0{
+                _offset + *s as isize * (*d as isize -1)
+            }else{
```

Comment:
```suggestion
            } else {
```

I won't mention all missing spaces in the PR I think :)
 
Still, please *don't* reformat changes outside the diffs that are relevant to this PR.

Reply:
Sorry for that. I will pay attention to it.